server/src/test/java/org/elasticsearch/snapshots/RestoreServiceTests.java

```diff
@@ -47,77 +47,112 @@
 public class RestoreServiceTests extends ESTestCase {
 
     public void testUpdateDataStream() {
+        long now = System.currentTimeMillis();
         String dataStreamName = "data-stream-1";
         String backingIndexName = DataStream.getDefaultBackingIndexName(dataStreamName, 1);
-        List<Index> indices = Collections.singletonList(new Index(backingIndexName, "uuid"));
+        List<Index> indices = List.of(new Index(backingIndexName, randomUUID()));
+        String failureIndexName = DataStream.getDefaultFailureStoreName(dataStreamName, 1, now);
+        List<Index> failureIndices = List.of(new Index(failureIndexName, randomUUID()));
 
-        DataStream dataStream = DataStreamTestHelper.newInstance(dataStreamName, indices);
+        DataStream dataStream = DataStreamTestHelper.newInstance(dataStreamName, indices, failureIndices);
 
         Metadata.Builder metadata = mock(Metadata.Builder.class);
-        IndexMetadata indexMetadata = mock(IndexMetadata.class);
-        when(metadata.get(eq(backingIndexName))).thenReturn(indexMetadata);
-        Index updatedIndex = new Index(backingIndexName, "uuid2");
-        when(indexMetadata.getIndex()).thenReturn(updatedIndex);
+
+        IndexMetadata backingIndexMetadata = mock(IndexMetadata.class);
+        when(metadata.get(eq(backingIndexName))).thenReturn(backingIndexMetadata);
+        Index updatedBackingIndex = new Index(backingIndexName, randomUUID());
+        when(backingIndexMetadata.getIndex()).thenReturn(updatedBackingIndex);
+
+        IndexMetadata failureIndexMetadata = mock(IndexMetadata.class);
+        when(metadata.get(eq(failureIndexName))).thenReturn(failureIndexMetadata);
+        Index updatedFailureIndex = new Index(failureIndexName, randomUUID());
+        when(failureIndexMetadata.getIndex()).thenReturn(updatedFailureIndex);
 
         RestoreSnapshotRequest request = new RestoreSnapshotRequest();
 
         DataStream updateDataStream = RestoreService.updateDataStream(dataStream, metadata, request);
 
         assertEquals(dataStreamName, updateDataStream.getName());
-        assertEquals(Collections.singletonList(updatedIndex), updateDataStream.getIndices());
+        assertEquals(List.of(updatedBackingIndex), updateDataStream.getIndices());
+        assertEquals(List.of(updatedFailureIndex), updateDataStream.getFailureIndices());
```

Comment:
Will these tests work if the feature flag is disabled? I recall that we enable the flag just about everywhere but I can't remember if there are still cases where the tests run without it on (like for releases).

Reply:
I thought the unit tests have always the feature flags one, isn't this the case?